### ISSUE TITLE
Making Cookies HTTPOnly

### ISSUE THREAD

**Issue by ChakshuGautam** [getodk/aggregate#502]
#### Software and hardware versions
macOS Docker Compose, Aggregate v2.0.5 [Self built WAR] behind NGINX reverse proxy

#### Problem description
Trying to make cookies secure by

```proxy_cookie_path / "/; HTTPOnly; Secure";```
But I see that `HTTPOnly` cannot be enabled due to `doc.cookie` API being used internally to cache.

#### Steps to reproduce the problem
Docker Compose is the with the default config. 
My current NGINX config. 

```nginx
server {
    listen 80;
     server_name aggregate.apple.server;
     return 301 https://$host$request_uri;
 }

 server {
    listen 443 ssl;
    listen [::]:443 ssl;
    server_name  aggregate.apple.server;

    # Timeout + DDoS
    client_body_timeout 3s;
    client_header_timeout 3s;

    # Limit requests size
    client_body_buffer_size 200K;
    client_header_buffer_size 2k;
    client_max_body_size 200k;
    large_client_header_buffers 3 1k;

    add_header X-Frame-Options "SAMEORIGIN";

    # HTTP only secure.
    proxy_cookie_path / "/; HTTPOnly; Secure";


    location / {
        proxy_pass http://server:8080/;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
    }

    ssl_certificate /root/certs/auth+4.pem;
    ssl_certificate_key /root/certs/auth+4-key.pem;

    error_page  404              /404.html;

    error_page   500 502 503 504  /50x.html;
    location = /50x.html {
        root   html;
    }
}
```
Aggregate will respond with an error => `You do not have permission for this action. Invalid Request`

#### Expected behavior
Should load normally.

**Comment by lognaturel** [comment]
Please note that the recommended ODK server is now [ODK Central](https://github.com/getodk/central) and the core ODK team no longer maintains Aggregate.

I believe you're referring to your own Tomcat config file and not a part of the Aggregate codebase. Have you tried changing that setting?

**Comment by ChakshuGautam** [comment]
Yes. The generated cache.js just stops working altogether. Not sure why. It has references to `doc.cookie` API which is disabled when we use `HttpOnly=true`.

**Comment by ChakshuGautam** [comment]
One of the reasons why we are not able to use ODK Central is webhooks. We have built a pipeline around webhooks which is still not supported by Central.

**Comment by lognaturel** [comment]
I would recommend searching [the forum](https://forum.getodk.org/) for setups similar to yours or posting if you don't find anything useful.

> We have built a pipeline around webhooks

It would be helpful to know more about your workflow. We've discussed some possibilities in this area but will need to have more user input before we build anything.
